sdk/storage/azure-storage-file-share/src/main/java/com/azure/storage/file/share/ShareFileAsyncClient.java

```diff
@@ -562,13 +552,17 @@ private boolean checkDoesNotExistStatusCode(Throwable t) {
 
         if (filePermissionCopyMode == null || filePermissionCopyMode == PermissionCopyModeType.SOURCE) {
             if (filePermission != null || filePermissionKey != null) {
-                throw logger.logExceptionAsError(new IllegalArgumentException(
+                return PollerFlux.error(LOGGER.logExceptionAsError(new IllegalArgumentException(
```

Comment:
is this change ok?

Reply:
Yes, this should be preferable to the exception escaping out of the reactive stream, and synchronous behavior will remain the same.